packages/ai/router.py

```
@@ -1193,6 +1258,20 @@ async def _try_one_provider(
+                    # so in-flight never leaks. Swallow errors — accounting
+                    # must not be able to fail a request that already worked.
+                    if _director is not None:
+                        try:
+                            _director.record_end(
+                                provider.provider_id,
+                                latency_ms=int((time.perf_counter() - started) * 1000),
+                                tokens=_tokens_for_director,
+                                status_code=_status_for_director,
+                            )
+                        except Exception:  # nosec B110 - accounting is best-effort
+                            pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)